src/static.js

```diff
@@ -0,0 +1,26 @@
+
+import React from 'react'
+import classnames from 'classnames'
+import cxs from 'cxs'
+
+const createComponent = (Tag = 'div') => (style = {}) => {
+  const cx = cxs(style)
+
+  const Component = ({ className, ...props }) => {
+    const combinedClassName = classnames(cx, className)
+
```

Comment:
You can further optimise performance here by calling Tag directly (instead of via React.createElement).

``` javascript
// stolen from recompose
const isReferentiallyTransparentFunctionComponent = Component => Boolean(
  typeof Component === 'function' &&
  Component.prototype.isReactComponent !== 'object' &&
  !Component.defaultProps &&
  !Component.contextTypes &&
  (process.env.NODE_ENV === 'production' || !Component.propTypes)
)

if (isReferentiallyTransparentFunctionComponent(Tag)) {
  return Tag(newProps);
} else  if(typeof Tag === 'string') { // Too far?
  return {type: Tag, props: newProps, $$typeof: Symbol.for('react.element')};
} else {
     return (
       <Tag
          {...newProps}
        />
      )
}
```